### PR TITLE
Starcoin main v6 vecslice

### PR DESCRIPTION
**Pull request by welbon**
## Motivation

Solve the problem of high gas cost caused by frequent copying of vector without slice function

### Have you read the [Contributing Guidelines on pull requests](https://github.com/move-language/move/blob/main/CONTRIBUTING.md#developer-workflow)?

(Write your answer here.)

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)

### CODE REVIEW

File: language/move-vm/types/src/values/values_impl.rs [hunk 1]
```diff
@@ -2245,6 +2245,42 @@ impl VectorRef {
...
+        memory_cost: &mut AbstractMemorySize,
+        offset: usize,
+        length: usize,
+        type_param: &Type,
+    ) -> PartialVMResult<Value> {
+        let c = self.0.container();
+        check_elem_layout(type_param, &c)?;
+        let container = match c {
+            Container::VecU8(r) => {
+                Value::vector_u8(r.borrow().get(offset..offset + length).unwrap().to_vec())
```
Comment (nkysg) [inline language/move-vm/types/src/values/values_impl.rs]: unwrap()可以处理成报错吗

Comment (welbon) [reply language/move-vm/types/src/values/values_impl.rs]: ok，这里我处理下

**Review by nkysg — APPROVED** [review]

File: language/move-vm/types/src/values/values_impl.rs (hunk 1 above)
Comment (nkysg) [reply language/move-vm/types/src/values/values_impl.rs]: 这里不知道用map_err会更好处理不， @sanlee42 俊哥帮忙看看

Comment (sanlee42) [reply language/move-vm/types/src/values/values_impl.rs]: map_err可以的，抛出来就行。

**Review by sanlee42 — APPROVED** [review]

**Pull request merged**